### PR TITLE
Feature/swift4.2

### PR DESCRIPTION
**Pull request by igorkislyuk**

### CODE REVIEW

File: Sources/PinLayout.swift [hunk 1]
```diff
@@ -22,9 +22,19 @@ import Foundation
 #if os(iOS) || os(tvOS)
     import UIKit
     public typealias PEdgeInsets = UIEdgeInsets
+
+    public extension UIEdgeInsets {
```
Comment (lucdion) [inline Sources/PinLayout.swift]: Please remove the `public` keyword. This should be kept private.

File: Sources/PinLayout.swift [hunk 2]
```diff
@@ -22,9 +22,19 @@ import Foundation
...
+        // using .zero crashes compiler. https://bugs.swift.org/browse/SR-7879
+        static let nonCrashZero = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+    }
 #else
     import AppKit
     public typealias PEdgeInsets = NSEdgeInsets
+
+    public extension NSEdgeInsets {
```
Comment (lucdion) [inline Sources/PinLayout.swift]: Same here.

File: Sources/PinLayout.swift (hunk 1 above)
Comment (lucdion) [reply Sources/PinLayout.swift]: In fact you should move these extensions inside files `Extensions/UIView+PinLayout` and `Extensions/NSView+PinLayout`.

**Comment by igorkislyuk** [comment]
Ok, sorry didn't receive this notification about response. Will do it sooner

**Comment by lucdion** [comment]
Finally the Beta 3 is out and fix the UIEdgeInsets.zero crash 🙂. See release notes https://download.developer.apple.com/Developer_Tools/Xcode_10_beta_3/Release_Notes_for_Xcode_10_beta_3.pdf

**Comment by igorkislyuk** [comment]
So this pull can be closed? Or we still need update for new Xcode/

**Comment by lucdion** [comment]
I've tested using Xcode 10 Beta 3, and PinLayout now works correctly. I'll close this PR. 
Thanks for fixing it anyway 👍

**Comment by igorkislyuk** [comment]
Thank you very much for taking your on this pull request. Glad that everything is ok now

**Pull request closed without merging**